test/arrayops.jl

```diff
@@ -3295,3 +3295,44 @@ end
     ref = memoryref(mem, 2)
     @test parent(ref) === mem
 end
+
+# some tests marked broken because CI runs with check-bounds=yes which impedes escape analysis
```

Comment:
Put it in https://github.com/JuliaLang/julia/blob/master/test/boundscheck_exec.jl or start a new process with the required flags?